packages/core/src/utils/styleUtils/ssrStyles.ts

```diff
@@ -596,13 +692,36 @@ export const indexByBreakpoint = ({
       // TODO: Test this for nested patterns as the name might be just a random hash without the actual name (needs to be validated).
       variableName.startsWith('cfBackgroundImageUrl_')
     ) {
+      const width = resolveImageOptimizationOptions({
+        variableData: variables['cfWidth'],
+        defaultBreakpoint,
+        componentSettings,
+        componentVariablesOverwrites,
+      }) as string;
+
+      const options = resolveImageOptimizationOptions({
+        variableData: variables['cfBackgroundImageOptions'],
+        defaultBreakpoint,
+        componentSettings,
+        componentVariablesOverwrites,
+      }) as BackgroundImageOptions;
```

Comment:
these typecasts are looking error prone, especially given that's the same function

Reply:
maybe separate functions, where each one should return a static type?

Reply:
Yeah, most of the logic in the function is same. So wanted it to be a single function.

Reply:
ok then maybe at least rename it to something like `resolveVariable` since it's a generic function